UE4SS/src/LuaType/LuaUEnum.cpp

```diff
@@ -110,16 +110,199 @@ No overload found for function 'UEnum.ForEachName'.
             return 0;
         });
 
-        if constexpr (is_final == LuaMadeSimple::Type::IsFinal::Yes)
-        {
-            table.add_pair("type", [](const LuaMadeSimple::Lua& lua) -> int {
-                lua.set_string(ClassName::ToString());
-                return 1;
-            });
-
-            // If this is the final object then we also want to finalize creating the table
-            // If not then it's the responsibility of the overriding object to call 'make_global()'
-            // table.make_global(ClassName::ToString());
-        }
+        table.add_pair("GetEnumNameByIndex", [](const LuaMadeSimple::Lua& lua) -> int {
+            std::string error_overload_not_found{R"(
+No overload found for function 'UEnum.InsertIntoNames'.
+Overloads:
+#1: GetNameByValue(integer Value))"};
+
+            auto& lua_object = lua.get_userdata<UEnum>();
+
+            if (!lua.is_integer())
+            {
+                lua.throw_error(error_overload_not_found);
+            }
+
+            auto value = lua.get_integer();
+            LuaType::FName::construct(lua, lua_object.get_remote_cpp_object()->GetEnumNameByIndex(value));
```

Comment:
This should ideally mimic `UEnum::GetEnumNameByIndex` by returning both the name and the value.
We could make use of multiple return values that Lua supports here by constructing both values and `return 2` instead of `return 1`.

Reply:
I believe this is setup correctly now. Please verify to see if it makes sense.

Reply:
Looks good.